flowmachine/flowmachine/features/subscriber/interevent_period.py

```diff
@@ -0,0 +1,146 @@
+
+    Examples
+    --------
+
+    >>> s = IntereventPeriod("2016-01-01", "2016-01-07")
+    >>> s.get_dataframe()
+
+        subscriber       interevent_period_avg
+        JZoaw2jzvK2QMKYX       03:26:47.673913
+        1QBlwRo4Kd5v3Ogz       03:23:03.979167
+        V4DJ7AaxLLOyWEZ3       03:39:35.977778
+        woOyJvzKBALYj8za       03:14:27.780000
+        1NqnrAB9bRd597x2       03:57:17.926829
+                     ...                   ...
+
+    """
+
+    def __init__(
+        self,
+        start,
+        stop,
+        statistic="avg",
+        *,
+        hours="all",
+        tables="all",
+        subscriber_identifier="msisdn",
+        subscriber_subset=None,
+        direction="both",
+    ):
+
+        self.start = start
+        self.stop = stop
+        self.hours = hours
+        self.tables = tables
+        self.subscriber_identifier = subscriber_identifier
+
+        if direction not in {"in", "out", "both"}:
+            raise ValueError("{} is not a valid direction.".format(self.direction))
+        self.direction = direction
+
+        if self.direction in {"both"}:
+            column_list = [self.subscriber_identifier, "datetime"]
+        elif self.direction in {"in", "out"}:
+            column_list = [self.subscriber_identifier, "datetime", "outgoing"]
+        else:
+            raise ValueError("{} is not a valid direction.".format(self.direction))
+
+        verify_columns_exist_in_all_tables(self.connection, self.tables, column_list)
+
+        self.statistic = statistic.lower()
+        if self.statistic not in valid_stats:
+            raise ValueError(
+                "{} is not a valid statistic. Use one of {}".format(
+                    self.statistic, valid_stats
+                )
+            )
+
+        self.unioned_query = EventsTablesUnion(
+            self.start,
+            self.stop,
+            tables=self.tables,
+            columns=column_list,
+            hours=self.hours,
+            subscriber_identifier=self.subscriber_identifier,
+            subscriber_subset=subscriber_subset,
+        )
+        super().__init__()
+
+    @property
+    def column_names(self):
+        return ["subscriber", f"interevent_period_{self.statistic}"]
+
+    def _make_query(self):
+
+        where_clause = ""
+        if self.direction != "both":
+            where_clause = (
+                f"WHERE outgoing IS {'TRUE' if self.direction == 'out' else 'FALSE'}"
+            )
+
+        # Postgres does not support the following three operations with intervals
+        if self.statistic in {"median", "stddev", "variance"}:
+            statistic_clause = (
+                f"MAKE_INTERVAL(secs => {self.statistic}(EXTRACT(EPOCH FROM delta)))"
+            )
+        else:
+            statistic_clause = f"{self.statistic}(delta)"
+
+        sql = f"""
+        SELECT
+            subscriber,
+            {statistic_clause} AS interevent_period_{self.statistic}
```

Comment:
Per #421 can we drop the suffix on this column?

Reply:
Ok, what would be the best to call it simply `interevent_period` or `value`? I didn't quite get from #421 what's our consensus for column names.

Reply:
I think eventually we want to have column names that are the same across all queries of a particular type (e.g. the two column names `subscriber` and `value` for queries that return a subscriber-based metric). This would eliminate case distinctions (and the need to pass column names as parameters) from any code that consumes a subscriber-based metric because it doesn't need to worry which column names the query returns. So personally I'd prefer using `value` here.

Reply:
@greenape, do you agree with max? Can I switch the column name to `value`?